examples/02_decoding/plot_haxby_different_estimators.py

```diff
@@ -117,8 +117,14 @@
 
 # Then we make a rudimentary diagram
 import matplotlib.pyplot as plt
+from matplotlib import __version__ as mpl_version
 
-plt.subplots(figsize=(8, 6), layout="constrained")
+from nilearn._utils import compare_version
+
+if compare_version(mpl_version, ">=", "3.5"):
+    plt.subplots(figsize=(8, 6), layout="constrained")
+else:
+    plt.subplots(figsize=(8, 6), constrained_layout=True)
 
```

Comment:
Ok, at first look, all these version checks in examples felt weird, but then I went through the older discussions about version bumps. Scikit-learn is still supporting 3.3 MPL so I guess we are sticking with that too.

Reply:
The technical writer in me is having second thoughts about these changes in the example.
I think they may distract the user from the point of the example: in the end those things are there mostly to remind maintainers to update this code when we bump matplotlib version: so I think I am going to stick with support of the 'old' matplotlib version and open an issue that will act as proper reminder.

Reply:
I agree that this overhead is quite bad in terms of clarity.

Reply:
- removed that from the examples and opened #4672 instead